projects/RabbitMQ.Client/client/impl/AutorecoveringModel.cs

```diff
@@ -1184,16 +1189,16 @@ public void BasicCancelNoWait(string consumerTag)
                 throw new ObjectDisposedException(GetType().FullName);
             }
 
-            string result = _delegate.BasicConsume(queue, autoAck, consumerTag, noLocal,
+            string resultConsumerTag = _delegate.BasicConsume(queue, autoAck, consumerTag, noLocal,
                 exclusive, arguments, consumer);
-            RecordedConsumer rc = new RecordedConsumer(this, queue).
-                WithConsumerTag(result).
+            RecordedConsumer rc = new RecordedConsumer(this, queue, resultConsumerTag).
```

Comment:
@Zerpet this is where I introduced the bug in a prior commit. I removed the code that sets the consumer tag, whoops. So, I changed the `RecordedConsumer` ctor to _require_ a tag.